src/plugins/boolean/boolean.c

```diff
@@ -0,0 +1,291 @@
+/**
+ * @file
+ *
+ * @brief Source for boolean plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "boolean.h"
+
+#include <ctype.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+#define DEFAULT_TRUE_VALUE "1"
+#define DEFAULT_FALSE_VALUE "0"
+
+
+typedef struct
+{
+	char * true;
+	char * false;
+	char ** trueValues;
+	char ** falseValues;
+} BoolData;
+
+int elektraBooleanClose (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	// free all plugin resources and shut it down
+	// this function is optional
+	BoolData * data = elektraPluginGetData (handle);
+	if (!data) return 1;
+	if (data->trueValues)
+	{
+		char ** ptr = (data->trueValues);
+		while (*ptr)
+		{
+			elektraFree (*ptr);
+			++(ptr);
+		}
+		elektraFree (data->trueValues);
+	}
+	if (data->falseValues)
+	{
+		char ** ptr = (data->falseValues);
+		while (*ptr)
+		{
+			elektraFree (*ptr);
+			++(ptr);
+		}
+		elektraFree (data->falseValues);
+	}
+	elektraFree (data);
+	elektraPluginSetData (handle, NULL);
+	return 1; // success
+}
+
+char * toUpper (const char * string)
+{
+	char * newString = strdup (string);
+	char * ptr = newString;
+	while (*ptr)
+	{
+		*ptr = toupper (*ptr);
+		++ptr;
+	}
+	return newString;
+}
+
+static int isTrue (const char * value, const char ** trueValues)
+{
+	char ** ptr = (char **)trueValues;
+	char * localString = toUpper (value);
+	int retVal = 0;
+	while (*ptr)
+	{
+		if (!strcmp (localString, *ptr))
+		{
+			retVal = 1;
+			break;
+		}
+		++ptr;
+	}
+	free (localString);
+	return retVal;
+}
+
+static int isFalse (const char * value, const char ** falseValues)
+{
+	char ** ptr = (char **)falseValues;
+	char * localString = toUpper (value);
+	int retVal = 0;
+	while (*ptr)
+	{
+		if (!strcmp (localString, *ptr))
+		{
+			retVal = 1;
+			break;
+		}
+		++ptr;
+	}
+	free (localString);
+	return retVal;
+}
+
+static void normalize (Key * key, BoolData * data)
+{
+	const char * defaultTrueValues[] = {
+		"TRUE", "1", "ON", NULL,
+	};
+	const char * defaultFalseValues[] = {
+		"FALSE", "0", "OFF", NULL,
+	};
+	const char * value = keyString (key);
+	const char ** falseValues = (const char **)data->falseValues;
+	const char ** trueValues = (const char **)data->trueValues;
+	const char ** falseStrings = falseValues ? falseValues : defaultFalseValues;
+	const char ** trueStrings = trueValues ? trueValues : defaultTrueValues;
+	if (isTrue (value, trueStrings))
+	{
+		keySetMeta (key, "origvalue", keyString (key));
+		keySetString (key, data->true);
+	}
+	else if (isFalse (value, falseStrings))
+	{
+		keySetMeta (key, "origvalue", keyString (key));
+		keySetString (key, data->false);
+	}
```

Comment:
Please add at least a warning if its neither true nor false.

It was suggested that true should be used in this case.